<launch>

    <group ns="uav0">
        <arg name="uav_0_camera_name" default="/airsim_node/PX4_0/downwards_custom_0"/> 
        <include file="$(find utm)/launch/broadcast_tf.launch">
            <arg name ="quad_tf" value="/uav0_wrt_world"/>
            <arg name ="world_tf" value="/world_enu"/>
            <arg name="uav_name" value="uav0"/>

            <arg name ="rtagdrone" value="/tag_wrt_uav0"/>
            <arg name="camera_name" value="$(arg uav_0_camera_name)"/>
            <arg name="frame_tag_id" value="/tag_0"/>

            <arg name="id_tag" value="0"/>
            <arg name="height_from_tag" value="1"/>
            <arg name="old_target_tf" value="downwards_custom_0_optical"/>
            <!--<arg name="old_target_tf" value="$(arg old_target_tf)"/>-->
        </include>
        <node pkg="utm" type="moving_avg_main" name="moving_avg_main"/>
    </group>

    <group ns="uav1">
        
        <arg name="uav_1_camera_name" default="/airsim_node/PX4_1/downwards_custom_1"/> 
        <include file="$(find utm)/launch/broadcast_tf.launch">
            <arg name ="quad_tf" value="/uav1_wrt_world"/>
            <arg name ="world_tf" value="/world_enu"/>
            <arg name="uav_name" value="uav1"/>

            <arg name ="rtagdrone" value="/tag_wrt_uav1"/>
            <arg name="camera_name" value="$(arg uav_1_camera_name)"/>
            <arg name="frame_tag_id" value="/tag_1"/> <!--looking for tag 1-->

            <arg name="id_tag" value="1"/> 
            <arg name="height_from_tag" value="1"/>
            <arg name="old_target_tf" value="downwards_custom_1_optical"/>
            <!--<arg name="old_target_tf" value="$(arg old_target_tf)"/>-->
        </include>
        <node pkg="utm" type="moving_avg_main" name="moving_avg_main"/>
    </group>

</launch>